### PR TITLE
*: add metrics for raftkv.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @BusyJay @disksing @zhangjinpeng1987

### CODE REVIEW

File: src/storage/engine/mod.rs [hunk 1]
```diff
@@ -44,12 +46,14 @@ pub trait Engine: Send + Debug {
     fn async_snapshot(&self, ctx: &Context, callback: Callback<Box<Snapshot>>) -> Result<()>;
 
     fn write(&self, ctx: &Context, batch: Vec<Modify>) -> Result<()> {
+        ENGINE_OP_COUNTER_VEC.with_label_values(&["write", ""]);
```
Comment (disksing) [inline src/storage/engine/mod.rs]: Miss `inc()`.

**Comment by siddontang** [comment]
PTAL

File: src/storage/engine/metrics.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            &["type"]
+        ).unwrap();
+
+    pub static ref ENGINE_OP_COUNTER_VEC: CounterVec =
+        register_counter_vec!(
+            "tikv_storage_engine_op_total",
+            "Total number of engine operators",
+            &["type", "cf"]
+        ).unwrap();
+
```
Comment (overvenus) [inline src/storage/engine/metrics.rs]: Remove the empty line 39

File: src/storage/engine/mod.rs [hunk 3]
```diff
@@ -44,12 +46,14 @@ pub trait Engine: Send + Debug {
     fn async_snapshot(&self, ctx: &Context, callback: Callback<Box<Snapshot>>) -> Result<()>;
 
     fn write(&self, ctx: &Context, batch: Vec<Modify>) -> Result<()> {
+        ENGINE_OP_COUNTER_VEC.with_label_values(&["write", ""]).inc();
```
Comment (overvenus) [inline src/storage/engine/mod.rs]: `""`, why left "cf" blank?

Comment (siddontang) [inline src/storage/engine/mod.rs]: we don't care cf here.

**Comment by disksing** [comment]
LGTM

**Comment by BusyJay** [comment]
Is there any benchmark that shows the performance change after adding these metrics?

**Comment by siddontang** [comment]
There are some benchmarks for the metic library itself, the metric vec is about 120ns/op, we can optimize this later, now it is ok. 
Of course, this may still reduce the performance, but I think now we should add as many metrics as we can, to help us verify the system. 

/cc @ngaut @disksing

**Comment by ngaut** [comment]
LGTM

**Pull request merged**